src/uct/sm/mm/base/mm_ep.c

```diff
@@ -190,7 +190,7 @@ static inline ucs_status_t uct_mm_ep_get_remote_elem(uct_mm_ep_t *ep, uint64_t h
     *elem = UCT_MM_IFACE_GET_FIFO_ELEM(iface, ep->fifo, elem_index);
 
     /* try to get ownership of the head element */
-    returned_val = ucs_atomic_cswap64(&ep->fifo_ctl->head, head, head+1);
+    returned_val = ucs_atomic_cswap64(ucs_unaligned_ptr(&ep->fifo_ctl->head), head, head+1);
```

Comment:
maybe we can declare head as aligned ptr? since it's actually aligned..

Reply:
it will help for head only, uct_mm_fifo_ctl is packed and access to signal_sockaddr will fail again

Reply:
BTW i see `struct uct_mm_fifo_ctl` is packed, why is it not good enough? what is the error from gcc?

Reply:
```
In file included from /home/sergeyo/ucx/src/ucs/sys/sys.h:16,
                 from /home/sergeyo/ucx/src/uct/base/uct_iface.h:18,
                 from /home/sergeyo/ucx/src/uct/base/uct_md.h:10,
                 from sm/mm/mm_def.h:9,
                 from sm/mm/mm_iface.h:11,
                 from sm/mm/mm_ep.h:11,
                 from sm/mm/mm_ep.c:8:
sm/mm/mm_ep.c: In function ‘uct_mm_ep_get_remote_elem’:
sm/mm/mm_ep.c:191:57: error: taking address of packed member of ‘struct uct_mm_fifo_ctl’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
  191 |     returned_val = ucs_atomic_cswap64(ucs_unaligned_ptr(&ep->fifo_ctl->head), head, head+1);
/home/sergeyo/ucx/src/ucs/sys/compiler.h:71:43: note: in definition of macro ‘ucs_unaligned_ptr’
   71 | #define ucs_unaligned_ptr(_ptr) ((void*)(_ptr))
      |                                           ^~~~
cc1: all warnings being treated as errors
In file included from /home/sergeyo/ucx/src/ucs/sys/sys.h:16,
                 from /home/sergeyo/ucx/src/uct/base/uct_iface.h:18,
                 from /home/sergeyo/ucx/src/uct/base/uct_md.h:10,
                 from sm/mm/mm_def.h:9,
                 from sm/mm/mm_iface.h:11,
                 from sm/mm/mm_iface.c:7:
sm/mm/mm_iface.c: In function ‘uct_mm_iface_create_signal_fd’:
sm/mm/mm_iface.c:433:60: error: taking address of packed member of ‘struct uct_mm_fifo_ctl’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
  433 |                       (struct sockaddr *)ucs_unaligned_ptr(&iface->recv_fifo_ctl->signal_sockaddr),
/home/sergeyo/ucx/src/ucs/sys/compiler.h:71:43: note: in definition of macro ‘ucs_unaligned_ptr’
   71 | #define ucs_unaligned_ptr(_ptr) ((void*)(_ptr))
      |                                           ^~~~
make[2]: *** [Makefile:1996: sm/mm/libuct_la-mm_ep.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
cc1: all warnings being treated as errors
make[2]: *** [Makefile:1989: sm/mm/libuct_la-mm_iface.lo] Error 1
make[2]: Leaving directory '/home/sergeyo/ucx/src/uct'
make[1]: *** [Makefile:686: all-recursive] Error 1
make[1]: Leaving directory '/home/sergeyo/ucx'
make: *** [Makefile:556: all] Error 2
```

Reply:
it seems unaligned ptr diagnostic becomes too paranoiac :)

Reply:
yeah, it even complains about 1st member of the struct..